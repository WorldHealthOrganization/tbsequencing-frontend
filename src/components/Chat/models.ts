export interface IUserMessage {
  email: string,
  username: string,
  firstName: string,
  lastName: string
}
