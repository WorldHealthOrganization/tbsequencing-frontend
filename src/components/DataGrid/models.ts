export interface IColumn<T> {
  name: keyof T;
  header: string;
}

export enum GridType {
  Gene = 'gene',
  Drug = 'drug',
}

export enum ScrollType {
  Infinite = 'infinite',
  Regular = 'regular',
}

export type SortOrder = 'asc' | 'desc' | undefined;

export interface ISortState {
  colIndex?: number
  order?: SortOrder
  colID?: string
}

export interface IColumnFilter {
  id: string;
  value: number | string;
}

export interface BaseColumns {
  variantName: string;
  variantGrade: string;
  consequence: string;
  globalFrequency: string;
  totalCounts: number;
  resistantCount: number;
  susceptbleCount: number;
  intermediateCount: number;
  nucleodicAnnName: string;
  proteicAnnName: string;
  drug: number;
  drugName: string;
}

export interface DrugTableColumns extends BaseColumns {
  geneName: string;
}

export interface GeneTableColumns extends BaseColumns {
  drug: number;
}
