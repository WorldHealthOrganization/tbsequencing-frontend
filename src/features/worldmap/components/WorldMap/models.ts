export type MapType = 'macroRegions' | 'heatmap';
