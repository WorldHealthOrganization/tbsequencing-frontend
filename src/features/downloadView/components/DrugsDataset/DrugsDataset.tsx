/** @jsxImportSource @emotion/react */
import React, { useRef, useEffect } from 'react';
import { CSVLink } from 'react-csv';
import { toast } from 'react-toastify';
import Checkbox from '@mui/material/Checkbox';
import FormControlLabel from '@mui/material/FormControlLabel';
import H3 from '../../../../components/typography/H3';
import PrimaryText from '../../../../components/typography/PrimaryText';
import AppButton from '../../../../components/AppButton';
import {
  wrapper, box, header, content, main, mainTop, list, asLink, asLinkFirst, control, csvLink,
} from './styles';
import { useExportTable } from '../../../../hooks/useExportTable';
import { IExtendedDrug } from '../../models';

interface DrugsDatasetProps {
  drugsList: IExtendedDrug[] | [];
  handleChange: (touchedItem: IExtendedDrug, checked: boolean) => void;
  handleCheckAll: (isSelect: boolean) => void;
  mockDate?: string;
}

const DrugsDataset = ({
  drugsList, handleChange, handleCheckAll, mockDate,
}: DrugsDatasetProps) => {
  // react-scv lib doesn't have types for ref
  const csvRef = useRef<any>(null);

  const getDrugIds = () => {
    let drugsIds = '';
    drugsList.forEach((drugItem) => {
      if (drugItem.isChecked) {
        drugsIds += `,${drugItem.drugId}`;
      }
    });
    return drugsIds.substring(1);
  };

  const {
    response,
    fetchData,
  } = useExportTable(
    {
      drugID: getDrugIds(),
      filters: [],
    },
  );

  useEffect(() => {
    if (csvRef?.current && response.data?.length && response.status === 'fulfilled') {
      csvRef.current.link.click();
    }
  }, [response]);

  useEffect(() => {
    if (response.data?.length === 0) {
      toast.info('There is no Dataset');
    }
  }, [response.data]);

  const getCSVFileName = () => {
    // use only in snapshot
    if (mockDate) return `tbkb-dataset-${mockDate}`;

    const currentDate = new Date();
    const day = currentDate.getDate();
    const month = currentDate.getMonth() + 1;
    const year = currentDate.getFullYear();
    return `tbkb-dataset-${year}-${month}-${day}`;
  };

  return (
    <div css={wrapper}>
      <div css={box}>
        <H3 style={header}>Download data</H3>
        <div css={content}>
          <PrimaryText>
            Please find below for each drug the updated allele counts and frequencies for related
            genes. The counts will reflect the current state of the database as new samples are
            added, and will be identical to those shown in the different interactive views of
            TBKB.
          </PrimaryText>
          <CSVLink
            css={csvLink}
            ref={csvRef}
            data={response.data || []}
            filename={getCSVFileName()}
          />
          <AppButton
            onClick={fetchData}
            variant="contained"
            size="medium"
            startIconName="file_download"
            disabled={response.isLoading || !drugsList.length}
          >
            {response.isLoading ? 'Loading...' : 'Download Dataset'}
          </AppButton>
        </div>
      </div>

      <div css={main}>
        <div css={mainTop}>
          <H3>Drugs</H3>
          <span css={[asLink, asLinkFirst]} onClick={() => handleCheckAll(true)} role="none">
            Select All
          </span>
          <span css={asLink} onClick={() => handleCheckAll(false)} role="none">
            Unselect All
          </span>
        </div>

        <div css={list}>
          {drugsList.map((drugItem) => (
            <FormControlLabel
              key={drugItem.drugId}
              value="end"
              control={(
                <Checkbox
                  onChange={() => handleChange(drugItem, drugItem.isChecked)}
                  checked={drugItem.isChecked}
                  css={control}
                  disabled={response.isLoading}
                />
              )}
              label={drugItem.drugName}
              labelPlacement="end"
            />
          ))}
        </div>

      </div>
    </div>
  );
};

export default DrugsDataset;
