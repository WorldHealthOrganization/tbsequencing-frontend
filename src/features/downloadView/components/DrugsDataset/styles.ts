import { css } from '@emotion/react';
import appColors from '../../../../styles/colors';

export const header = css({
  fontWeight: 'bold',
});

export const wrapper = css({
  color: appColors.neutral.black,
  fontSize: 16,
});

export const box = css({
  padding: '16px 24px',
  boxShadow: `0px 0px 4px ${appColors.tints.whoBlue.lighter}`,
  borderRadius: '4px 4px 0 0',
});

export const content = css({
  marginTop: 4,
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
});

export const main = css({
  boxShadow: `0px 2px 4px ${appColors.tints.whoBlue.lighter}`,
  padding: '30px 40px 36px 24px',
  borderRadius: '0 0 4px 4px',
});

export const asLink = css({
  textDecoration: 'underline',
  padding: '6px 16px',
  fontSize: 16,
  color: appColors.secondary.whoBlue,
  ':hover': {
    cursor: 'pointer',
  },
});

export const asLinkFirst = css({
  marginLeft: 'auto',
  marginRight: '20px',
});

export const mainTop = css({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  marginBottom: 24,
});

export const list = css({
  display: 'grid',
  gridTemplateColumns: '1fr 1fr 1fr 1fr 1fr',
  gap: '20px',
});

export const control = css({
  '&.MuiCheckbox-root': {
    color: appColors.secondary.whoBlue,
  },
});

export const csvLink = css({
  visibility: 'hidden',
});
