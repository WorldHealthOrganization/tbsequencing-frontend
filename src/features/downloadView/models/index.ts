import { IDrug } from '../../../services/drugsApi/models';

export interface IExtendedDrug extends IDrug {
  isChecked: boolean;
}
