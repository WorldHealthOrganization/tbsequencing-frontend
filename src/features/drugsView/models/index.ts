export enum DrugStatus {
  Susceptible = 'susceptible',
  Resistant = 'resistant',
  Intermediate = 'intermediate',
  RatioSusceptible = 'ratioSusceptible',
  RatioResistant = 'ratioResistant',
  RatioIntermediate = 'ratioIntermediate',
}

export enum ResistantType {
  Phenotypic = 'Pheno',
  Genotypic = 'Geno',
}

export enum ChartDataType {
  Ratio = 'ratio',
  Count = 'count',
}

export enum Regions {
  ALL = 'ALL_REGIONS',
  AFRICA = 'AFR',
  EUROPE = 'EUR',
  AMERICA = 'AMR',
  EASTERN_MEDITERRANEAN = 'EMR',
  WESTERN_PACIFIC = 'WPR',
  SOUTH_EAST_ASIA = 'SEA',
}
