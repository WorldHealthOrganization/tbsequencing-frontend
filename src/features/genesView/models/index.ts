import { skipToken } from '@reduxjs/toolkit/query';
import React from 'react';
import { IGene, IVariant } from '../../../services/genesApi/models';

export type AutocompleteParams = typeof skipToken | { search?: string };

export interface IGenomeContextWrapperProps {
  data?: IGene[];
  isLoading?: boolean;
  selectedGene?: IGenesOverview;
  minDomain?: number;
  maxDomain?: number;
}

export interface IGenomeContextProps extends IGenomeContextWrapperProps {
  width: number;
}

export interface IVariantLocal extends IVariant {
  pos: number;
}

export interface ISearchResult {
  geneName: string;
  geneDbCrossrefId: number;
  locusTag: string;
}

export interface ISearchCount extends ISearchResult {
  counter: number;
}

export interface ISearchRecent extends ISearchResult {
  counter: string;
  date: Date | string;
}

export interface IGenesOverview extends IGene {
  geneDescription: string;
  geneType: string;
  ncbiId: string;
  proteinLength: number;
}

export interface IGenesOverviewTableData {
  title: string;
  value: string | number | null;
  element?: React.ReactNode;
}

export interface IGenesOverviewResponse {
  genes: IGene;
  genesOverview: IGenesOverview
}
