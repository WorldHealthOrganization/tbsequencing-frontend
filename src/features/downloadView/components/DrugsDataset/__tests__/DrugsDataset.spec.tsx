import React from 'react';
import {
  getTreeJSONWithRedux, render, screen, fireEvent,
} from '../../../../../utils/testUtils/jestUtils';
import DrugsDataset from '../DrugsDataset';
import { IExtendedDrug } from '../../../models';

describe('DrugsDataset', () => {
  const mockDrugsList: IExtendedDrug[] = [
    {
      drugId: 1, drugName: 'Isoniazid', code: 'INH', isChecked: false,
    },
    {
      drugId: 2, drugName: 'Rifampicin', code: 'RIF', isChecked: false,
    },
    {
      drugId: 3, drugName: 'Streptomycin', code: 'STR', isChecked: false,
    },
    {
      drugId: 4, drugName: 'Ethambutol', code: 'EMB', isChecked: false,
    },
    {
      drugId: 5, drugName: 'Capreomycin', code: 'CAP', isChecked: false,
    },
  ];

  const handleChange = jest.fn();
  const handleCheckAll = jest.fn();
  const props = {
    drugsList: mockDrugsList,
    handleChange,
    handleCheckAll,
    mockDate: '2023-04-21',
  };
  test('renders DrugsDataset component without error', () => {
    render(
      <DrugsDataset {...props} />,
    );
  });

  test('calls correct callbacks Download Dataset Type', () => {
    render(
      <DrugsDataset {...props} />,
    );

    const btn = screen.getByText('Select All');
    fireEvent.click(btn);

    expect(handleCheckAll).toHaveBeenCalledTimes(1);
  });

  test('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <DrugsDataset {...props} />,
    );

    expect(tree).toMatchSnapshot();
  });
});
