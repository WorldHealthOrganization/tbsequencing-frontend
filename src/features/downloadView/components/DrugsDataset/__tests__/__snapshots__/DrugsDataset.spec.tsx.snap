// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrugsDataset Should render correctly 1`] = `
<div
  className="css-dxqlm-DrugsDataset"
>
  <div
    className="css-1628001-DrugsDataset"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-15lj30p-MuiTypography-root-BaseText"
    >
      Download data
    </p>
    <div
      className="css-98h9wr-DrugsDataset"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
      >
        Please find below for each drug the updated allele counts and frequencies for related genes. The counts will reflect the current state of the database as new samples are added, and will be identical to those shown in the different interactive views of TBKB.
      </p>
      <a
        className="css-h7lobx-DrugsDataset"
        download="tbkb-dataset-2023-04-21"
        href="data:text/csv;charset=utf-8,ï»¿"
        onClick={[Function]}
        target="_self"
      />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1ve9vu2-MuiButtonBase-root-MuiButton-root-AppButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <span
            aria-hidden={true}
            className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            file_download
          </span>
        </span>
        <p
          className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
        >
          Download Dataset
        </p>
      </button>
    </div>
  </div>
  <div
    className="css-xxefzu-DrugsDataset"
  >
    <div
      className="css-1c1h8c7-DrugsDataset"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-5th3yp-MuiTypography-root-BaseText"
      >
        Drugs
      </p>
      <span
        className="css-1dy1e9u-DrugsDataset"
        onClick={[Function]}
        role="none"
      >
        Select All
      </span>
      <span
        className="css-1p35x0h-DrugsDataset"
        onClick={[Function]}
        role="none"
      >
        Unselect All
      </span>
    </div>
    <div
      className="css-1oc9pf2-DrugsDataset"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-cqwuvp-MuiButtonBase-root-MuiCheckbox-root-DrugsDataset"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Isoniazid
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-cqwuvp-MuiButtonBase-root-MuiCheckbox-root-DrugsDataset"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Rifampicin
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-cqwuvp-MuiButtonBase-root-MuiCheckbox-root-DrugsDataset"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Streptomycin
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-cqwuvp-MuiButtonBase-root-MuiCheckbox-root-DrugsDataset"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Ethambutol
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-cqwuvp-MuiButtonBase-root-MuiCheckbox-root-DrugsDataset"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Capreomycin
        </span>
      </label>
    </div>
  </div>
</div>
`;
